### PR TITLE
chore: update core to 0.19, api to rc0

### PR DESCRIPTION
**Pull request by dyladan**
For now I am keeping the API dependency where it is and updating it to latest. a follow up will move it to peer deps.

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-aws-lambda/package.json [hunk 1]
```diff
@@ -63,10 +63,10 @@
     "typescript": "3.9.7"
   },
   "dependencies": {
-    "@opentelemetry/api": "^0.18.1",
-    "@opentelemetry/instrumentation": "^0.18.2",
-    "@opentelemetry/resources": "^0.18.2",
-    "@opentelemetry/semantic-conventions": "^0.18.2",
-    "@opentelemetry/tracing": "^0.18.2"
+    "@opentelemetry/api": "^1.0.0-rc.0",
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/package.json:66]: I think we should move API to peer dependencies first to stay consistent with core.

Comment (dyladan) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/package.json:66]: I am trying to debug an issue here. that's why its a draft. my local npm install is failing.

Comment (Flarna) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/package.json:66]: npm 6 or 7?

Comment (dyladan) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/package.json:66]: 6.14.11

```
lerna ERR! npm install exited 1 in '@opentelemetry/instrumentation-redis'
lerna ERR! npm install stderr:
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @opentelemetry/semantic-conventions@^0.19.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'opentelemetry-instrumentation-redis'
npm ERR! notarget 
```

Happens on all packages that depend on semantic conventions

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/aws-lambda.ts [hunk 2]
```diff
@@ -120,8 +120,8 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
       const span = plugin.tracer.startSpan(name, {
         kind: SpanKind.SERVER,
         attributes: {
-          [FaasAttribute.FAAS_EXECUTION]: context.awsRequestId,
-          [FaasAttribute.FAAS_ID]: context.invokedFunctionArn,
+          [SemanticAttributes.FAAS_EXECUTION]: context.awsRequestId,
+          'faas.id': context.invokedFunctionArn,
```
Comment (dyladan) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/src/aws-lambda.ts:124]: faas.id is used as a span attribute here, but should be a resource attribute. I left it here to make this PR as small as possible while passing and changing no behavior. I will create a follow up bug to fix this.

**Review by Flarna — APPROVED** [review]

**Comment by lizthegrey** [comment]
exciting!

**Review by naseemkullah — APPROVED** [review]

**Review by lizthegrey — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Review by obecny — APPROVED** [review]
lgtm

**Pull request merged**